### ISSUE TITLE
Polynomial factorization over number fields

### ISSUE THREAD

**Issue by jankoboehm** [oscar-system/Oscar.jl#2434]
Factorization of a polynomial over a non-simple number field yields a stack overflow error:

```
Qt, t = QQ["t"];
K, (a1,a2) = number_field([t^2 - 2, t^2 - 3], "a");
R, (x,y) = polynomial_ring(K,["x","y"]);
f = x^2-6
factor(f)
```

Should this give a not implemented?

Error message:

```
ERROR: StackOverflowError:
Stacktrace:
     [1] factor(a::AbstractAlgebra.Generic.MPoly{NfAbsNSElem})
       @ Hecke ~/.julia/packages/Hecke/ttSxw/src/Misc/Integer.jl:712
     [2] factor(x::AbstractAlgebra.Generic.MPoly{NfAbsNSElem})
       @ Hecke ~/.julia/packages/Hecke/ttSxw/src/Hecke.jl:239
     [3] factor(a::AbstractAlgebra.Generic.MPoly{NfAbsNSElem}; b::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
       @ Hecke ~/.julia/packages/Hecke/ttSxw/src/Misc/Integer.jl:712--- the last 3 lines are repeated 26660 more times ---
```

System:

```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.12.2-DEV
  combining:
    AbstractAlgebra.jl   v0.30.7
    GAP.jl               v0.9.6
    Hecke.jl             v0.18.14
    Nemo.jl              v0.34.4
    Polymake.jl          v0.9.2
    Singular.jl          v0.18.3
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.6+0
    GAP_jll                 v400.1200.200+1
    Singular_jll            v403.201.1+0
    libpolymake_julia_jll   v0.9.2+0
    libsingular_julia_jll   v0.30.0+1
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.9.0
Commit 8e630552924 (2023-05-07 11:25 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin22.4.0)
  CPU: 16 × Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, skylake)
  Threads: 1 on 16 virtual cores
Environment:
  JULIA_IMAGE_THREADS = 1
Official https://julialang.org/ release
```

**Comment by fieker** [comment]
On Tue, May 30, 2023 at 09:32:34AM -0700, Janko Boehm wrote:
> Factorization of a polynomial over a non-simple number field yields a stack overflow error:
> 
> ```
> Qt, t = QQ["t"];
> K, (a1,a2) = number_field([t^2 - 2, t^2 - 3], "a");
> R, (x,y) = polynomial_ring(K,["x","y"]);
> f = x^2-6
> factor(f)
> ```
> 
That should be fixed - won't be fast, but should be possible.

**Comment by fieker** [comment]
On Tue, May 30, 2023 at 09:32:34AM -0700, Janko Boehm wrote:
> Factorization of a polynomial over a non-simple number field yields a stack overflow error:
> 
> ```
> Qt, t = QQ["t"];
> K, (a1,a2) = number_field([t^2 - 2, t^2 - 3], "a");
> R, (x,y) = polynomial_ring(K,["x","y"]);
> f = x^2-6
> factor(f)
> ```
> 
"fixed" in Hecke pr #1097
